src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs

```diff
@@ -328,7 +328,14 @@ protected virtual void OnTextChanged(DependencyPropertyChangedEventArgs e)
 
 			var focusManager = VisualTree.GetFocusManagerForElement(this);
 			if (focusManager?.FocusedElement != this &&
-				GetBindingExpression(TextProperty) is { ParentBinding.UpdateSourceTrigger: UpdateSourceTrigger.Default or UpdateSourceTrigger.LostFocus } bindingExpression)
+				GetBindingExpression(TextProperty) is
+				{
+					ParentBinding:
+					{
+						IsXBind: false, // NOTE: we UpdateSource in OnTextChanged only when the binding is not an x:Bind. WinUI's generated code for x:Bind contains a simple LostFocus subscription and waits for the next LostFocus even when not focused, unlike regular Bindings.
```

Comment:
NOTE: WinUI's generates LostFocus only if UpdateSourceTrigger is LostFocus (which is the default for TextBox.Text specifically). But if you explicitly set UpdateSourceTrigger on an x:Bind for TextBox.Text to something other than LostFocus, the generated code will be different.

Reply:
To be clear, only if UpdateSourceTrigger is LostFocus OR Default